gu-client/src/async.rs

```diff
@@ -1,20 +1,28 @@
+use crate::error::Error;
 use actix_web::{client, http, HttpMessage};
 use bytes::Bytes;
-use error::Error;
-use futures::stream::Stream;
-use futures::{future, Future};
+use futures::{future, prelude::*};
 use gu_actix::release::{AsyncRelease, Handle};
 use gu_model::peers::PeerInfo;
 use gu_model::{
+    deployment::DeploymentInfo,
     envman,
-    session::{self, BlobInfo, HubSessionSpec, Metadata},
+    session::{self, BlobInfo, HubExistingSession, HubSessionSpec, Metadata},
 };
+use gu_net::rpc::peer::PeerSessionInfo;
 use gu_net::types::NodeId;
-use std::str;
+use gu_net::types::TryIntoNodeId;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::borrow::Borrow;
+use std::collections::VecDeque;
 use std::sync::Arc;
 use std::time::Duration;
+use std::{env, str};
 use url::Url;
 
+pub type HubSessionRef = Handle<HubSession>;
+
```

Comment:
would make `HubConnection` (below) 1-uple instead of struct with single `hub_connection_inner`